KappaAnalysis/interface/Producers/ValidJetsProducer.h

```diff
@@ -604,54 +604,98 @@ class ValidJetsProducerBase: public KappaProducerBase, public ValidPhysicsObject
             }
             else if (jetIDVersion == KappaEnumTypes::JetIDVersion::ID2017UL || jetIDVersion == KappaEnumTypes::JetIDVersion::ID2018UL) {
                 if (std::abs(jet->p4.eta()) <= 2.6f) {
-                    float maxNHFraction = 0.90f;
-                    float maxNEMFraction = 0.90f;
-                    float minNumConstituents = 1;
-                    float maxMuFraction = 0.8f;
-                    float minCHFraction = 0;
-                    float minChargedMult = 0;
-                    float maxCEMFraction = 0.80f;
+                    if (jetID == KappaEnumTypes::JetID::TIGHTLEPVETO) {
+                        float maxNHFraction = 0.90f;
+                        float maxNEMFraction = 0.90f;
+                        float minNumConstituents = 1;
+                        float maxMuFraction = 0.8f;
+                        float minCHFraction = 0;
+                        float minChargedMult = 0;
+                        float maxCEMFraction = 0.80f;
 
-                    if (!(jet->neutralHadronFraction < maxNHFraction &&
-                        (jet->photonFraction + jet->hfEMFraction) < maxNEMFraction &&
-                        jet->nConstituents > minNumConstituents &&
-                        jet->muonFraction < maxMuFraction &&
-                        jet->chargedHadronFraction > minCHFraction &&
-                        jet->nCharged > minChargedMult &&
-                        jet->electronFraction < maxCEMFraction))
-                        return false;
+                        if (!(jet->neutralHadronFraction < maxNHFraction &&
+                              (jet->photonFraction + jet->hfEMFraction) < maxNEMFraction &&
+                              jet->nConstituents > minNumConstituents &&
+                              jet->muonFraction < maxMuFraction &&
+                              jet->chargedHadronFraction > minCHFraction &&
+                              jet->nCharged > minChargedMult &&
+                              jet->electronFraction < maxCEMFraction))
```

Comment:
... and this to an additional fanout, which only covers `tightlepveto` case.

And do the same below.